packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -193,7 +193,7 @@ const devHtmlHook: IndexHtmlTransformHook = async (
     )
   }
 
-  await traverseHtml(html, htmlPath, (node) => {
+  await traverseHtml(html, filename, (node) => {
```

Comment:
This `htmlPath` argument is only passed to `handleParseError`, so it's safe to change.
https://github.com/vitejs/vite/blob/e58a4f00e201e9c0d43ddda51ccac7b612d58650/packages/vite/src/node/plugins/html.ts#L158-L172